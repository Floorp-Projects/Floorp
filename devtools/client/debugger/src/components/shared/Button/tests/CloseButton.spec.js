/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */

import React from "react";
import { shallow } from "enzyme";
import { CloseButton } from "../";

describe("CloseButton", () => {
  it("renders with tooltip", () => {
    const tooltip = "testTooltip";
    const wrapper = shallow(
      <CloseButton tooltip={tooltip} handleClick={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("handles click event", () => {
    const handleClickSpy = jest.fn();
    const wrapper = shallow(<CloseButton handleClick={handleClickSpy} />);
    wrapper.simulate("click");
    expect(handleClickSpy).toHaveBeenCalled();
  });
});
