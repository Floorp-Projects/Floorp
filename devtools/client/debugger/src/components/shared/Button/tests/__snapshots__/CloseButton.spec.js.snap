// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CloseButton renders with tooltip 1`] = `
<button
  className="close-btn"
  onClick={[Function]}
  title="testTooltip"
>
  <AccessibleImage
    className="close"
  />
</button>
`;
